### PR TITLE
recipe_gdal

### PR DESCRIPTION
**Pull request by drewgilliam**
GDAL recipe - CentOS 6 with GDAL 3+, including PROJ v6, ECW J2K 5.5, & OpenJPEG 2.3
Usage in another image: `COPY --from=gdal /gdal/usr/local /usr/local`

Recipe based on the centos6 manylinux2010 docker image.  This image already has updated build essentials necessary for compilation of GDAL & its dependencies, and offers a path in the future to compile portable GDAL (including appropriate python bindings & all dependencies in one nice package).

Notes:
- Includes general test of gdal (e.g., `gdalinfo --version` returns expected version). 
- docker-compose update to v2.3 to deal with multi-stage builds.  
- Compatible with `ci_load` as per recent PRs for fast CI builds (as long as you didn't change the GDAL dockerfile)
- The manylinux2010 base image is pinned to a particular build to support `ci_load` caching

### CODE REVIEW

File: recipe_gdal.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+# This dockerfile follows procedures from the offical GDAL dockers
+#   https://github.com/OSGeo/gdal/tree/master/gdal/docker
+#
+# This dockerfile is derived from the manylinux2010 base image, derived from
+# CentOS 6 and already containing many updated build essentials.
+#   https://github.com/pypa/manylinux
+#
+# For the future, the manylinux2010 image may enables a possible portable GDAL
+# that includes internal copies of necessary dependencies and python bindings
+# for a selected python version.
```
Comment (scott-vsi) [inline recipe_gdal.Dockerfile:15]: What version of python are you building for here? I don't see it specified

(also, no 's' on enables)

Comment (drewgilliam) [reply recipe_gdal.Dockerfile:15]: There are no python bindings built here (yet), that's a future idea.  Users will still get their python bindings from pypi.  I'm updating the comments to make this more clear with a working example.

File: README.rst [hunk 2]
```diff
@@ -323,6 +323,55 @@ https://github.com/onetrueawk/awk is a severly limited version awk that some pri
...
+   FROM vsiri/recipe:gdal as gdal
+   FROM ubuntu:16.04
+
+   # set shell to bash
+   SHELL ["/usr/bin/env", "/bin/bash", "-euxvc"]
+
+   # install python & gdal
+   COPY --from=python /usr/local /usr/local/
+   COPY --from=gdal /gdal/usr/local /usr/local
+   RUN ldconfig
```
Comment (andyneff) [inline README.rst:351]: I know it seems silly, but should this be put into a ["patch"](https://github.com/VisionSystemsInc/docker_recipes/blob/master/recipe_pipenv.Dockerfile#L8)? I'm doing `git lfs install` for git-lfs, a similar one line patch

Comment (drewgilliam) [reply README.rst:351]: I can add that for those who want it, but maybe the example shouldn't require the VSI recipe too?

**Pull request merged**